marquez/client.py

```diff
@@ -0,0 +1,264 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import urllib
+
+
+from marquez.constants import JobRunState
+from marquez.utils import request_utils
+from marquez.utils.request_utils import (InvalidRequestException,
+                                         handle_response)
+
+
+class MarquezClient(object):
+    API_PATH = "api/v1"
+    DEFAULT_TIMEOUT_SEC = 5
```

Comment:
When using the `requests` lib, only `seconds` is supported, do we need `DEFAULT_TIMEOUT_MILLISECONDS` if we are just going to convert to seconds?

Reply:
It's a float in the library, so if a user wants greater precision, they can have it.

```
 :param timeout: (optional) How many seconds to wait for the server to send data
        before giving up, as a float, or a :ref:`(connect timeout, read
        timeout) <timeouts>` tuple.
  ```

Reply:
Right, but it's made explicit that it will be expected as `seconds`.  We only need to set the timeout value with the assumption will be in seconds. The conversion is unnecessary. Meaning:

```python
DEFAULT_TIMEOUT = 5
```

Reply:
Prefer to have `DEFAULT_TIMEOUT_MS` to be explicit since we're dealing w/different units.